bgpd/bgp_zebra.c

```diff
@@ -3856,12 +3883,16 @@ static void bgp_zebra_process_srv6_locator_delete_per_bgp(struct srv6_locator *l
 		}
 	}
 
-	// refresh tovpn_sid
+	// refresh tovpn_sid and tovpn_sid_locator
 	for (ALL_LIST_ELEMENTS_RO(bm->bgp, node, bgp_vrf)) {
 		if (bgp->inst_type == BGP_INSTANCE_TYPE_VRF && bgp != bgp_vrf)
 			continue;
 
-		if (bgp_vrf->inst_type != BGP_INSTANCE_TYPE_VRF)
+		tovpn_sid_locator = bgp_srv6_locator_lookup(bgp_vrf, bgp);
```

Comment:
`bgp_srv6_locator_lookup` attempts to access `bgp->srv6_locator`, which has already been freed at line 3864:

https://github.com/FRRouting/frr/blob/73bfff4080190d469412e3dc0c40e9992cff603c/bgpd/bgp_zebra.c#L3862-L3866

To resolve this issue, we can move the "clear SRv6 locator" block to the end of the function, just before the call to `vpn_leak_postchange_all`.

Reply:
This change will likely solve the test failures as well.